File: ConflictsServer/src/main/java/islab/project/conflictsserver/conflict/Conflict.java
package islab.project.conflictsserver.conflict;

import islab.project.conflictsserver.conflict.converter.ConflictIntensity;
import islab.project.conflictsserver.conflict.converter.ConflictType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "conflict")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Conflict {
    @Id
    private Integer id;
    private String location;
    private String sideA;
    private String sideB;
    private ConflictType type;
    private ConflictIntensity intensity;
    private LocalDate startTime;
    private LocalDate endTime;

}

File: ConflictsWeb/src/models/Conflict.ts
import {number, string} from "yup";

export class Conflict {
    id: number;
    location: string;
    sideA: string;
    sideB: string;
    conflictType: string;
    conflictIntensity: string;
    startTime: string;
    endTime: string|undefined;


    constructor(id: number, location: string, sideA: string, sideB: string, conflictType: string, conflictIntensity: string, startTime: string, endTime: string | undefined) {
        this.id = id;
        this.location = location;
        this.sideA = sideA;
        this.sideB = sideB;
        this.conflictType = conflictType;
        this.conflictIntensity = conflictIntensity;
        this.startTime = startTime;
        this.endTime = endTime;
    }
}